doc/manual/metaprogramming.rst

```diff
@@ -134,19 +162,88 @@ ambiguity in parsing.:
     julia> :(::)
     :(::)
 
-:obj:`Symbol`\ s can also be created using :func:`symbol`, which takes
-a character or string as its argument:
+Expressions and evaluation
+--------------------------
+
+Quoting
+~~~~~~~
+
+There is special syntax to create expression objects without using the
+explicit :obj:`Expr` constructor above. A short form for inline expressions
+uses the ``:`` followed by a single parenthesized expression. Here is an
+example of the short form used to quote an arithmetic expression:
 
 .. doctest::
 
-    julia> symbol('\'')
-    :'
+    julia> ex = :(a+b*c+1)
+    :(a + b * c + 1)
 
-    julia> symbol("'")
-    :'
+    julia> typeof(ex)
+    Expr
+
+(to view the structure of this expression, try ``ex.head`` and ``ex.args``,
+or use :func:`dump` as above).
 
-:func:`eval` and Interpolation
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Note that equivalent expressions may be constructed using :func:`parse` or
+the direct :obj:`Expr` form:
+
+.. doctest::
+
+    julia>      :(a + b*c + 1)  ==
+	   parse("a + b*c + 1") ==
+	   Expr(:call, :+, :a, Expr(:call, :*, :b, :c), 1)
+    true
+
+Expressions provided by the parser generally only have symbols, other
+expressions, and literal values as their args, whereas expressions
+constructed by Julia code can have arbitrary run-time values
+without literal forms as args. In this specific example, ``+`` and ``a``
+are symbols, ``*(b,c)`` is a subexpression, and ``1`` is a literal
+64-bit signed integer.
+
+There is a second form of quoting for multiple expressions:
```

Comment:
"syntactic form"? 

Maybe it would be good to say here that `:(a + b)` and `quote a + b end` are not equivalent at the expression level.  Also, `quote` blocks introduce `QuoteNode`'s into the AST which is a bit confusing to newcomers.  That is a bit of the internal IR leaking through but you quickly run into it if you work with the AST of parsed julia source code.